app/src/main/java/com/bunbeauty/papakarlo/feature/address/screen/create_address/CreateAddressFragment.kt

```diff
@@ -18,51 +18,98 @@ import androidx.compose.ui.Modifier
-            val streetListState by viewModel.streetListState.collectAsState()
+            val streetListState by viewModel.streetListState.collectAsStateWithLifecycle()
             CreateAddressScreen(streetListState)
+            LaunchedEffect(streetListState.eventList) {
+                handleEventList(streetListState.eventList)
+            }
         }
     }
 
     @Composable
-    private fun CreateAddressScreen(streetListState: State<List<StreetItem>>) {
-        when (streetListState) {
-            is State.Success -> CreateAddressSuccessScreen(streetListState.data)
-            is State.Loading -> LoadingScreen()
-            is State.Error -> ErrorScreen(streetListState.message) {
-                viewModel.getStreetList()
+    private fun CreateAddressScreen(createAddressState: CreateAddressState) {
+        when (val state = createAddressState.state) {
+            is CreateAddressState.State.Success -> CreateAddressSuccessScreen(createAddressState)
+            is CreateAddressState.State.Loading -> LoadingScreen()
+            is CreateAddressState.State.Error -> {
+                when (state.throwable) {
+                    is GetStreetsUseCase.NoSelectedCityUuidThrow, CreateAddressUseCase.NoSelectedCityUuidThrow -> {
+                        ErrorScreen(
+                            mainTextId = R.string.error_create_address_no_selected_city,
+                            extraTextId = R.string.error_create_address_no_selected_city_help
+                        ) {
+                            viewModel.getStreetList()
+                        }
+                    }
+                    is GetStreetsUseCase.NoUserUuidThrow, CreateAddressUseCase.NoUserUuidThrow -> {
+                        ErrorScreen(
+                            mainTextId = R.string.error_create_address_no_selected_city,
+                            extraTextId = R.string.error_create_address_no_selected_city_help
+                        ) {
+                            viewModel.getStreetList()
+                        }
+                    }
+                    is CreateAddressUseCase.NoStreetByNameAndCityUuidThrow -> {
+                        ErrorScreen(
+                            mainTextId = R.string.error_create_address_no_street_by_city_uuid_and_name,
+                            extraTextId = R.string.error_create_address_no_street_by_city_uuid_and_name_help
+                        ) {
+                            viewModel.getStreetList()
+                        }
+                    }
+                    else -> {
+                        ErrorScreen(
+                            mainTextId = R.string.common_error,
+                            extraTextId = R.string.internet_error
+                        ) {
+                            viewModel.getStreetList()
+                        }
+                    }
+                }
+            }
+            is CreateAddressState.State.Empty -> {
```

Comment:
I think it's better use State.Error if we then show ErrorScreen

Reply:
+